Tests/GitHubCodespaces.tests.ps1

```diff
@@ -0,0 +1,443 @@
+                $codespace.git_status.ref | Should -Be $pullRequest.head.ref
+                $codespace.repository.name | Should -Be $repoWithPR.name
+                $codespace.owner.UserName | Should -Be $script:OwnerName
+                $codespace.template | Should -BeNullOrEmpty
+            }
+
+            AfterAll {
+                if ($codespace)
+                {
+                    Remove-GitHubCodespace -CodespaceName $codespace.name -Confirm:$false
+                }
+            }
+        }
+
+        Context -Name 'When creating a codespace with default settings from a PullRequest' {
+            BeforeAll {
+                  $repoWithPR = Get-GitHubRepository -OrganizationName $script:organizationName |
+                      Where-Object { $_ | Get-GitHubPullRequest } |
+                      Select-Object -First 1
+                $pullRequest = $repoWithPR | Get-GitHubPullRequest | Select-Object -First 1
+                $newGitHubCodespaceParms = @{
+                    PullRequest = $pullRequest.number
+                    RepositoryId = $repoWithPR.Id
+                }
+                $codespace = New-GitHubCodespace @newGitHubCodespaceParms
+                Start-Sleep -Seconds 2
+            }
+
+            It 'Should return an object of the correct type' {
+                $codespace | Should -BeOfType PSCustomObject
+            }
+
+            It 'Should return the correct properties' {
+                $codespace.display_name | Should -Not -BeNullOrEmpty
+                $codespace.repository.name | Should -Be $repoWithPR.name
+                $codespace.owner.UserName | Should -Be $script:OwnerName
+                $codespace.pulls_url | Should -Be $pullRequest.url
+                $codespace.template | Should -BeNullOrEmpty
+            }
+
+            AfterAll {
+                if ($codespace)
+                {
+                    Remove-GitHubCodespace -CodespaceName $codespace.name -Confirm:$false
+                }
+            }
+        }
+
+        Context -Name 'When creating a codespace with all possible settings' {
+            BeforeAll {
+                $newGitHubCodespaceParms = @{
+                    # ClientIp = 'TODO ???? - should be instead of rather than in addition to Geo, perhaps add some param validation to the function'
+                    # DevContainerPath = 'Will add to test in the future when Get-GitHubDevContainer is implemented and the test repo includes one'
+                    DisplayName = 'PowerShellForGitHub pester test'
+                    Geo = 'UsWest'
+                    Machine = 'basicLinux32gb'
+                    NoMultipleRepoPermissions = $true # Not sure how to assert this, but this proves it accepts the switch without error
+                    IdleRetentionPeriodMinutes = 10
+                    TimeoutMinutes = 5
+                    # WorkingDirectory = 'TODO ???? - not sure how to handle this'
```

Comment:
The real way to do it would be to mock out `Invoke-GHRestMethod` and just validate that the JSON body has the expected value, but all of the other tests that we do here are full end-to-end tests against the real API.  So, I wouldn't worry about this one either.